graphql_compiler/tests/integration_tests/test_backends_integration.py

```diff
@@ -421,10 +421,10 @@ def test_recurse(self):
         )
     )
     @integration_fixtures
-    def test_fold_basic(self, backend_name):
+    def test_fold_basic(self, backend_name: str) -> None:
         # (query, args, expected_results) tuples.
         # The queries are ran in the order specified here.
-        queries = [
+        queries: List[Tuple[str, Dict[str, Any], List[Dict[str, Any]]]] = [
```

Comment:
Could we re-use the `expected_results` type here to make this a bit more readable?

Reply:
I'm leaning toward typing first, refactoring as necessary later. Ideally, we'd get type hints everywhere, then take a look at how to streamline and improve the user-friendliness of the public and user-facing functions (e.g. with named typedefs), and then use those typedefs consistently throughout the codebase. For example, it's not clear to me on the basis of the integration tests file whether `QueryParameters = Dict[str, Any]` as a typedef would help or harm readability of the user-facing code.